docs/docs/guides/core-types/media.md

```diff
@@ -171,3 +171,118 @@ This audio is logged to Weave and automatically displayed in the UI, along with
 :::tip
 Try our cookbook for [Audio Logging](/reference/gen_notebooks/audio_with_weave) or <a href="https://colab.research.google.com/github/wandb/weave/blob/master/docs/./notebooks/audio_with_weave.ipynb" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link button button--secondary button--med margin-right--sm notebook-cta-button"><div><img src="https://upload.wikimedia.org/wikipedia/commons/archive/d/d0/20221103151430%21Google_Colaboratory_SVG_Logo.svg" alt="Open In Colab" height="20px" /><div>Open in Colab</div></div></a>. The cookbook also includes an advanced example of a Real Time Audio API based assistant integrated with Weave.
 :::
+
+## Files (PDFs, CSVs)
+
+<Tabs groupId="programming-language" queryString>
+  <TabItem value="python" label="Python" default>
+  
+  ### PDFs
+  You can log and visualize various file types like [PDFs](#pdfs) and [CSVs](#csvs) using the `weave.File` object. This includes PDFs with varying lengths or content. Weave captures file metadata like path, MIME type, and size in traces. You can also [preview](#preview-a-file) or [download](#download-a-file) traced files from the **Traces** tab.
+
+  ```python
+  import weave
+  from pathlib import Path
+
+  weave.init("pdf-example")
+
+  @weave.op
+  def return_file_pdf() -> weave.File:
+      # Returns a PDF file "example.pdf" as a Weave file object.
+      # Modify the filename/path as needed
+      file_path = Path(__file__).parent.resolve() / "assets" / "example.pdf"
+      return weave.File(file_path)
+
+  @weave.op
+  def accept_file_pdf(val: weave.File) -> str:
+      # Accepts a Weave file object and prints metadata to the trace.
+      # Useful for debugging or file inspection in ops.
+      print("Path:", val.path)
+      print("MIME type:", val.mimetype)
+      print("Size:", val.size)
+      return f"File size: {val.size}"
+
+  # Run the file through the accept_file_pdf function to inspect it
+  accept_file_pdf(return_file_pdf())
+  ```
+   
+  ![Trace view of `accept_file_pdf`](imgs/accept-file-pdf.png)
+
+  You can also work with multi-page PDFs:
+  
+  ```python
+  @weave.op
+  def return_file_pdf_many_pages() -> weave.File:
+      # Similar to the above, but returns a multi-page PDF as a Weave file object.
+      # Replace the file path with your own multi-page document.
+      file_path = Path(__file__).parent.resolve() / "assets" / "example-many-pages.pdf"
+      return weave.File(file_path)
+
+  @weave.op
+  def accept_file_pdf_many_pages(val: weave.File) -> str:
+      # Upload and trace a multi-page PDF file in Weave and log its metadata.
+      print("Path:", val.path)
+      print("MIME type:", val.mimetype)
+      print("Size:", val.size)
+      return f"File size: {val.size}"
+
+  accept_file_pdf_many_pages(return_file_pdf_many_pages())
+  ```
+
+  ### CSVs
```

Comment:
I would consider restructuring the Files section to first talk about attaching any arbitrary file, perhaps using a CSV as an example, but it could be anything. For arbitrary files you just get the captured metadata in the UI and the ability to download. For PDFs logged with this mechanism specifically we also have added the ability to preview in-app. And we may extend that to future filetypes in the future.